src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -166,12 +162,14 @@ pub trait MetadataChainExt: MetadataChain {
     where
         E: Error + From<IterBlocksError>,
     {
-        let mut decisions = vec![Decision::Next; visitors.len()];
+        let mut decisions = vec![MetadataVisitorDecision::Next; visitors.len()];
```

Comment:
Hm, I thought that the recent bug where visitor was crashing on unexpected event type showed us that `Decision::Next` is not a great default.

If a visitor only wants to handle events of certain type - the `Decision::Next` means that it can still receive events of any type during the first call, thus its logic has to be more complex to handle them...

Should we extend visitor trait with `initial_decision() -> MetadataVisitorDecision` function?

Reply:
> Should we extend visitor trait with initial_decision() -> MetadataVisitorDecision function?

Definitely -- it'll make our lives easier + code will be more declarative